meilisearch/index.py

```diff
@@ -228,7 +229,8 @@ def get_stats(self):
             )
         )
 
-    def search(self, query, opt_params=None):
+    # pylint: disable=dangerous-default-value
+    def search(self, query, opt_params={}):
```

Comment:
It's a very bad practice to declare your default value in your function definition.
When you declare a default value there, the `opt_params` is set as a "global var" and is defined only at the definition not on each call.

See the following code:
```shell
>>> def test(key, opt_params={}):
...     opt_params[key] = True
...     return opt_params
...
>>> test('hey')
{'hey': True}
>>> test('yo')
{'hey': True, 'yo': True}
>>>
```

You should follow the pep8 guideline and make a default param at `None` and set it after the start of the function.

Reply:
Wooo it does not do what I expected at all 😂 
Thanks @mde-patch, changing that!!

Reply:
Fixed with #97

Reply:
@eskombro @bidoubiwa, pretty sure you are interested in that! 😉

Reply:
We had a discussion about it with @bidoubiwa a few days/weeks ago. I can't find it. We finally decided to leave it as `None` in the definition, and do a condition inside of the function, initializing an empty dict if `opt_params` was set to None. We didn't find a better solution

Reply:
> We had a discussion about it with @bidoubiwa a few days/weeks ago. I can't find it. 

Indeed, I haven't seen it all all... Don't know where you discussed about that 😆